applications/new_schema.py

```diff
@@ -138,9 +135,7 @@ class Query:
         "\n* A value passed is not a valid status",
     )
 
-    @login_required
-    @superuser_required
-    # TODO: Should check if the user has permissions to access these objects
+    @view_permission_required(BerthApplication, BerthSwitch, HarborChoice)
```

Comment:
our GQL types are very intertwined. from `BerthApplicationNode` you can access `ProfileNode` and from `ProfileNode` you can access `BoatNode` and `CompanyType` 🤔

Reply:
Then we can replace it for:
```suggestion
    @view_permission_required(BerthApplication, CustomerProfile)
```

So `BerthApplication` covers `Application`, `Switch` and `HarborChoice` (since they all share the same perms) and `CustomerProfile` that covers `Boat` and `Company`.

Reply:
yes, let's follow this pattern